src/common/utility.ts

```diff
@@ -566,6 +566,47 @@ export class Utility {
         return path.join(solutionPath, Constants.vscodeFolder, Constants.vscodeSettingsFile);
     }
 
+    public static async validateRepositoryUrl(repositoryUrl: string): Promise<string | undefined> {
+        if (!repositoryUrl) {
+            return "Repository url could not be empty";
+        }
+
+        const match: RegExpMatchArray = repositoryUrl.match(/^(?:([^\/]+)\/)?([^:]+)(?::(.+))?$/);

```

Comment:
where did we get the regex or write by yourself?

Reply:
Can we just get the first part before "/" like what we did in getting address?

Reply:
myself, this regex is to divide the repository url to three part: hostName, repository name and tag.
However, the validate function cannot cover all invalid cases such as aa/bb///ccdd

Reply:
> Can we just get the first part before "/" like what we did in getting address?

Yes, the first part of this regex : (?:([^\/]+)\/)?  is to get first part before "/" just like we use indexOf("/")